src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java

```diff
@@ -66,6 +69,54 @@ public RequestResponsePact executeGetListOfRolesAndGet200(PactDslWithProvider bu
                 .toPact();
     }
 
+    @Pact(provider = "am_role_assignment_service", consumer = "am_org_role_mapping")
+    public RequestResponsePact executeCreateRoleAssignmentReplacingExistingFalseAndGet201(PactDslWithProvider builder) {
+
+        return builder
+                .given("The assignment request is valid with replaceExisting flag as false")
+                .uponReceiving("role assignment service takes s2s/auth token and create a role assignment")
+                .path(RAS_CREATE_ROLE_ASSIGNMENT_URL)
+                .method(HttpMethod.POST.toString())
+                .body(createRoleAssignmentRequestReplaceExistingFalse(), String.valueOf(ContentType.JSON))
+                .willRespondWith()
+                .status(HttpStatus.CREATED.value())
+                .headers(getRoleAssignmentResponseHeaders())
+                .body(createRoleAssignmentResponseReplaceExistingFalse())
+                .toPact();
+    }
+
+    @Pact(provider = "am_role_assignment_service", consumer = "am_org_role_mapping")
+    public RequestResponsePact executeCreateRoleAssignmentOneRoleAndGet201(PactDslWithProvider builder) {
+
+        return builder
+                .given("The assignment request is valid with one requested role")
+                .uponReceiving("role assignment service takes s2s/auth token and create or update a role assignment")
+                .path(RAS_CREATE_ROLE_ASSIGNMENT_URL)
+                .method(HttpMethod.POST.toString())
+                .body(createRoleAssignmentRequestOneRole(), String.valueOf(ContentType.JSON))
+                .willRespondWith()
+                .status(HttpStatus.CREATED.value())
+                .headers(getRoleAssignmentResponseHeaders())
+                .body(createRoleAssignmentResponseOneRole())
+                .toPact();
+    }
+
+    @Pact(provider = "am_role_assignment_service", consumer = "am_org_role_mapping")
+    public RequestResponsePact executeCreateRoleAssignmentZeroRoleAndGet201(PactDslWithProvider builder) {
+
+        return builder
+                .given("The assignment request is valid with zero requested role")
```

Comment:
We can provide more details here about this test
e.g. "The assignment request is valid with zero requested role and replaceExisting flag as true"

Reply:
Done